Update yz tests/zl/add deletes to crdt yz [JIRA: RIAK-2484]

zeeshanlakhani opened the pull request:
Test CRDTs (update), `delete_mode=keep`, handling for work in batching/deletes/yz fixes: https://github.com/basho/yokozuna/pull/565 / flush_queues.

lordnull committed:
Added test for realtime helper process leak.

lordnull committed:
Made test more reliable to check for leaks.

rpc or other processes will cause one or two persistent processes to start up during
the test, but the number shoould not increase after that. Altered the test to check for
massive increase in processes over a larger number of faults rather than an equality
of process_count over a relatively small number of faults.

russelldb committed:
Add failing test for github issue riak-727

russelldb committed:
Clean up comment typo and poorly named function

russelldb committed:
Remove extraneous and inaccurate comment

Commit:
Hopefully fix the disconnection issue in the riak667_mixed test

Commit:
Update build versions to 2.1.1

Commit:
Switch to using devrels instead of stagedevrels

Commit:
Add in the VERSION file for each version of the built devrel

Commit:
Change the VERSION file to have only the dotted version, not the product

Commit:
Strip the newline at the end of the VERSION file

zeeshanlakhani committed:
* clean-up yokozuna_rt for reusability
* add new types for dialyzer
* add test to reload schema and overwrite it on type change
* test nested search (e.g. nested json objects)
* test re-putting fields, include crdt map workaround, discussed in
https://gist.github.com/sdebnath/36c235e042cb35db7d1f
* increase aae concurrency for core_prop test

fadushin committed:
Re-write of handle_handoff_write_once to test handle_handoff_command.

Still needs some work!

fadushin committed:
Reverted to original verify_handoff_write_once test with modifications

* Backed out the change to riak_core_handoff_sender_intercepts which had
  a modified intercept
* Modified the handle_command intercept to ensure we don't count the
  vnode crashes that were happening before the fix
* Reverted the test to the original, but check to make sure we can read
  all the entries, including the ones we added during handoff

zeeshanlakhani committed:
update some yokozuna tests from work done in riak/2.0 branch for 2.0.6 release

fadushin committed:
Tried to make the verify_handoff_write_once test more predictable

fadushin committed:
Added a cap to the number of messages to be send asynchronously, to give handoff a chance to complete.

fadushin committed:
Merge pull request #808 from basho/zl/forward-merge-riak/2.0-yoko-tests

update some yokozuna tests from work done in riak/2.0 branch for 2.0.…

fadushin committed:
Force calls to riak_kv_vnode:handle_handoff_command during a handoff.

fadushin committed:
Cleaned up the verify_handoff_write_once to run faster
(by not waiting for handoffs, just pending changes, as the old test did)

fadushin committed:
Backed out a false start in the riak_core_handoff_sender intercept

fadushin committed:
Merge commit '62335cf416dd3d68b0fae524e78a4973d65da8fe' into bugfix/fd/RIAK-1850

fadushin committed:
Cleaned up async process to remove code that is not exercised by this test.

Commit:
Add basic cluster converge test to try to capture issues with cluster
convergence outside of other tests.

jvoegele committed:
Add verify_snmp_repl riak_test

The verify_snmp_repl test reproduces the error condition (one cluster
configured for realtime replication with two or more other clusters) and
passes if and only if the riak_snmp_stat_poller module correctly handles
this configuration.

https://bashoeng.atlassian.net/browse/RIAK-1884
https://github.com/basho/riak_snmp/issues/27

jvoegele committed:
Remove extraneous lager call

fadushin committed:
Added intercepts to count the synchronous and asynchronous calls through handle_command, to ensure we are testing the case statements in riak_kv_vnode:handle_command with write_once puts.

Commit:
Merge pull request #813 from basho/feature/simple_cluster_test

Add basic cluster converge test to try to capture issues with cluster convergence outside of other tests.

Commit:
Update documented versions to be 1.4, 2.0 and 2.1

Commit:
Back out addition of --ignore-removal from git add command

JeetKunDoug committed:
Merge pull request #814 from basho/bugfix/RIAK-1884

Add verify_snmp_repl riak_test

fadushin committed:
Write-Once test improvements
* Made tests of handoff conditions more strict than they were before
* Added an intercept to the async write_once test to test that asynchronous writes actually happened
* Added a test case for the riak_kv (advanced) allow_async_put configuration variable.

fadushin committed:
Added some documentation to the test.

Also changed the name of the internal identifier for the globally registered bg process, to avoid potential name collisions in the future.

fadushin committed:
Missed a change in previous commit

fadushin committed:
Mode a change in the protocol between vnodes and our bg proc in the test to keep vnodes in the handoff state until all writes have completed.

Plus some miscellaneous tightening of test conditions.

Commit:
Add product name back to VERSION file

fadushin committed:
Modified test to wait to send messages until all waiting vnodes are in the handoff state.  Also corrected some of the arithmetic errors in counting the number of entries we were supposed to send.  (doh)

fadushin committed:
Cleaned up commented code.  No change in behavior.

Commit:
Adjust verify_membackend:get_used_space/2 to work with Riak 2.1+

Commit:
Fix riak_control and riak_control_authentication for Riak 2.1+
- Use the actual version string (e.g. riak_ee-2.1.2) of devrel
  instead of "current", "previous", "legacy" to determine which
  tests to run
- Add rt:get_version/1 to find the version of devrels other than
  "current"

Commit:
Switch check_put_delete/1 to use a range of acceptable sizes

Commit:
Add rt:wait_until_bucket_type_visible to avoid floppiness in test - verify would fail because, while handoff had completed, the bucket type information hadn't yet been gossiped to the new node.

Commit:
Merge pull request #816 from basho/bugfix/bch/update_membackend_regex

Patch verify_membackend test to run in 2.1+

fadushin committed:
Merge pull request #812 from basho/bugfix/fd/RIAK-1850

Test fix for bad_match during handoff with write_once buckets

Commit:
Merge pull request #791 from basho/bugfix/sdc/667-mixed-auto-reconnect

Hopefully fix the disconnection issue in the riak667_mixed test

Commit:
Update the URL for pulling down kerl

zeeshanlakhani committed:
Merge pull request #800 from basho/add-tests/zl/test-reload-schema-field-funniness

Add new test for schema reload/overwrite

Commit:
If Darwin, don't build hipe/odbc

Commit:
Change verify_memorybackend to use calls to core and kv to determine
memory used, instead of a brittle parsing of data structures

zeeshanlakhani committed:
add some clarity to ordering of socket close and such

Commit:
Make the release build process more reliable

Many changes here:
* Only download source once from the Interwebs
* Skip a directory if it's already built
* Use locked-deps when appropriate

Commit:
Merge branch 'feature/bch/update-rtdev-versions' of github.com:/basho/riak_test into feature/bch/update-rtdev-versions

engelsanchez committed:
First EQC test for secondary indexing

It needs more love to match the functionality of the regular 2i tests,
but this is a good beginning. It can show a race in 2i pagination that
has been in the code for years. The race does not show up in run in a
single node with very few vnodes. I tried 32.  Ideally we could use
PULSE to make scheduling in the riak nodes less deterministic and help
find this kind of issue, but currently PULSE is a single node joint.

This test uses a different bucket for each EQC command run. It is not
perfect isolation, but it is good enough and very fast compared to
taking down the entire cluster in between each run.

(cherry picked from commit fa493858373642b4219273fcae3f98cd34f1c7d1)

Commit:
Make the erlang builds more official

Commit:
Merge pull request #819 from basho/feature/mra/rtdev-osx

If we're building on OS X, don't build HiPE or ODBC connector

Commit:
Build Basho-patched Erlang and add tweaks for CentOS

Commit:
Update R16 version to OTP_R16B02_basho9 tag

Commit:
Make release tags BASH variable

Commit:
Update versions built

Commit:
Merge pull request #815 from basho/feature/bch/update-rtdev-versions

Update Documentation and Build Scripts for Riak 2.1

zeeshanlakhani committed:
switcheroo

fadushin committed:
Merge pull request #824 from basho/bugfix/zl/big-dummy-blame-zeeshan-bucket-index-swap

switcheroo

Commit:
Merge pull request #817 from basho/bugfix/bch/update_riak_control_versions

Fix riak_control and riak_control_authentication for Riak 2.1+

zeeshanlakhani committed:
Merge pull request #820 from basho/tests/zl/yz-extractors-bit-o-cleanup-but-cant-repro-flappiness

add some clarity to ordering of socket close and such

engelsanchez committed:
Clean up and improve

General clean up and organization, with an eye on making this a
template for other EQC tests.

Preparing code to handle more than the PB client.

(cherry picked from commit 02b5e2128d98679c893e800103a5e580b1f81e84)

macintux committed:
Merge pull request #782 from basho/rdb/gh-riak-727

Add failing test for github issue riak-727

engelsanchez committed:
Fleshed out but disabled http client support

The http client seems to have a lot of problems and works differently
than the PB client.  Index values are not coming back on a get, even
though a PB get returns them, so indexing is not working properly.
Binaries are not encoded properly when the query index URL is created,
and empty binaries are taken to mean "no value".

(cherry picked from commit 1c29f7fb43cbc45753d999a160f9d359445fbf50)

Commit:
Merge pull request #821 from basho/bugfix/bch/cleanup_verify_membackend

Update verify_memorybackend for RIAK-2061

zeeshanlakhani committed:
Merge remote-tracking branch 'origin/riak/2.0' into merge-forward/zl/riak/2.0tomaster

Commit:
Merge pull request #833 from basho/merge-forward/zl/riak/2.0tomaster

Merge forward/zl/riak/2.0tomaster

Commit:
Merge branch 'riak/2.0'

zeeshanlakhani committed:
Merge remote-tracking branch 'origin/riak/2.0' into merge-forward/zl/riak/2.0_head_97be67233aaac3242a58d06700bf0fbe55619d80-to-2.1

zeeshanlakhani committed:
Merge pull request #838 from basho/merge-forward/zl/riak/2.0_head_97be67233aaac3242a58d06700bf0fbe55619d80-to-2.1

Merge forward/zl/riak/2.0 head 97be67233aaac3242a58d06700bf0fbe55619d80 to 2.1

ksauzz committed:
Move rt_cs_dev to cs repo

Commit:
Forward merge (potential) fix for riak667_mixed {error, disconnected} issue.

fadushin committed:
Merge pull request #843 from basho/bugfix/dr/forward_merge_riak667_connection_fix

Forward merge riak667 connection fix to master

Commit:
Added rt:create_activate_and_wait_for_bucket_type so that the test can
specify it wants all nodes in the cluster to have the type before
continuing. Update http_bucket_types to use said function.

fadushin committed:
Merge pull request #847 from basho/bugfix/dr/http_bucket_types_should_wait_for_visibility

Added rt:create_activate_and_wait_for_bucket_type

jvoegele committed:
Fail with the bare atom fail

`riak_test_runner` does not support tagged tuples for failures. Fix
verify_snmp_repl module so that it returns the bare atom `fail` when it
fails, and write more detailed error messages to the log instead.

jvoegele committed:
Crash test process instead of returning fail atom

ksauzz committed:
Merge pull request #844 from basho/feature/move-cs-harness-to-cs-repo

Move rt_cs_dev to cs repo

jvoegele committed:
Log when original version of intercepted function fails

jvoegele committed:
Merge pull request #852 from basho/bugfix/verify-snmp-repl-fail-appropriately

Fail by crashing test process instead of returning a tagged tuple

Commit:
Merge branch 'riak/2.0' into forward_merge/dr/riak_2.0_to_master
- Forward merge verify_stage_clustering fix
- Updates to allow either strings or atoms for node names in config & allow them to be different (atom or string) in test vs. config.

JeetKunDoug committed:
Merge pull request #854 from basho/forward_merge/dr/riak_2.0_to_master

Forward merge/dr/riak 2.0 to master

jvoegele committed:
Induce race condition in riak_snmp_stat_poller

There is a race in riak_snmp_stat_poller, which occurs when realtime
replication is enabled for more than two clusters. If the replication is
enabled quickly enough, then the bug does not manifest itself. This
change introduces a sleep between enabling replication between the
individual clusters, thus causing the test to fail consistently instead
of sporadically.

zeeshanlakhani committed:
update extractors test to check for possible issue w/ expiry and custom extractors

zeeshanlakhani committed:
update yz tests to use commit fn instead of sleep

zeeshanlakhani committed:
test deletions w/ yz_crdt + tombstones + delete_mode -> keep

jvoegele committed:
Verify cluster stats in SNMP

Update verify_snmp_repl test to prove that cluster stats are correctly
recorded in SNMP.

mikewallace1979 committed:
Add copyright and Apache License Version 2.0

Commit:
Merge pull request #864 from mikewallace1979/add-apache-licese-to-intercepts-files

Add copyright and Apache License Version 2.0

zeeshanlakhani committed:
handle deletes via aae cases

zeeshanlakhani committed:
extend crdt test w/ partition/siblings/vtag test and make sure it behaves properly w/ new tombstone code in yz

jvoegele committed:
Refactor some Yokozuna tests to eliminate duplication

Specifically, introduce the `gen_keys/1` and `assert_search/6` functions
into yokozuna_rt module and refactor tests to use them instead of copied
and pasted code fragments.

(cherry picked from commit 49d5c361c51e73b18757961437dcf382b6d162ea)

zeeshanlakhani committed:
update intercepts for change to delete_operation/5 in yz branch->features+fixes/zl/handle-cleanup-allow_mult=true

zeeshanlakhani committed:
yokozuna_rt additions for types and better testing w/ allow-mult=true extractors test

jvoegele committed:
Merge pull request #866 from basho/bugfix/forward-merge-snmp-stat-poller-race

Merge branch `riak/2.0` into `master` w/ some updates to yz_extractors along the way

Commit:
Merge of several YZ changes and overload fix from riak/2.0 to riak/2.1

JeetKunDoug committed:
Merge pull request #901 from basho/bugfix/merge-overload-from-2.0

Forward merge 2.0 to 2.1 for Overload, and pick up some YZ fixes as well

JeetKunDoug committed:
Merge pull request #905 from basho/merge/2.1-to-master

Merge/2.1 to master

zeeshanlakhani committed:
sure up possible race between upgrade and first check

(cherry picked from commit 867dc1a0364e8a19e1a929dd9f3385f46703cfc4)

zeeshanlakhani committed:
Merge pull request #934 from basho/forwardport/zl/sure-up-yz_extractors

sure up possible race between upgrade and first check

nickelization committed:
Fix race in repl_rt_heartbeat due to short timeout

One particular timeout in the repl_rt_heartbeat test was slightly too
short, which could cause us to occasionally hit a false positive on this
test if various timings lined up just right. This PR bumps up the
timeout, which should prevent this from happening again.

I would really like to do a proper fix for this, which would use
intercepts or something to confirm that the actual timeout is being hit
in the code...but we don't really have time for that, and a half fix is
better than no fix I suppose.

nickelization committed:
Merge pull request #937 from basho/bugfix/heartbeat-timeout

Fix race in repl_rt_heartbeat due to short timeout

ksauzz committed:
Add cleanup intercepts function

This function purges specified target module, which was created as a proxy,
and its backuped original module, then reloads original module.

ksauzz committed:
Update spec

shino committed:
Merge pull request #961 from basho/feature/add-cleanup-intercepts

Add cleanup intercepts fuction

lukebakken committed:
Syntax error in .riak_test.config

Commit:
Ensure git adds everything and ignores any global ignore settings

Commit:
Merge pull request #980 from basho/fixes/lrb/force-add-allthethings

Ensure git adds everything and ignores any global ignore settings

Commit:
Prevent large negative integers as values

This is due to a sext encoding bug present in current riak releases.
When sext and riak has been updated with a fix, revert this commit.

(cherry picked from commit b5a0dd9737935a39c55f544409b7cf9e96ff66a0)

Commit:
Always ensure leveldb is the backend for 2i

(cherry picked from commit 472fb6447bea41d13ca823048b644a5d4f11d205)

Commit:
Add tests for read repair object limits

Tests to verify that object limits are ignored for read repair. This
requires an intercept to spoof all puts as read repair since there’s no
external API for forcing read repair. Intercept used to replace both
put/6 and coord_put/6 and simply adds the `rr` property to the put
options.

Commit:
Merge pull request #722 from basho/bug/mw/rt-helper-process-leak

Added test for realtime helper process leak.

nickelization committed:
Remove unnecessary sleeps in verify_object_limits

Discussed this with Brian and turned out they were just left over from
some debugging he'd been doing earlier. I tried the test without them
and it works fine.

nickelization committed:
Merge pull request #1009 from basho/feature/bjs/readrepair_object_limit

Add tests for read repair object limits

nickelization committed:
Merge pull request #1012 from basho/nem/forward-merge-2i-eqc

Forward merge new verify_2i_eqc test to master

Commit:
Merge pull request #1021 from basho/bch-merge-master-to-develop

Merge master changes over to develop

Commit:
Merge pull request #1022 from basho/bch-merge-develop-to-2.2

Merge the latest from develop to develop-2.2

JeetKunDoug commented:
create jira issue

Basho-JIRA commented:
Assuming this is ready for review/merge once the related work is done. 

***[posted via JIRA by Douglas Rohrer]***

Basho-JIRA commented:
[~zlakhani] Please provide a progress update 

***[posted via JIRA by Patricia Brewer]***

Commit:
Merge remote-tracking branch 'origin/riak/2.0' into bch-merge-2.0-to-develop-2.2

Commit:
Fix merge issues with overload.erl - re-instate w1c tests.

JeetKunDoug committed:
Merge pull request #1061 from basho/bch-merge-2.0-to-develop-2.2

Merge riak/2.0 to develop-2.2

zeeshanlakhani committed:
- new test for HLL(set) datatypes
- update crdt cap and converge tests for hlls
- temp rebar updates for riak-client; *will update w/ correct version*
upon mergies

Commit:
Merge remote-tracking branch 'origin/riak/2.0' into bch-merge-2.0-onto-develop-2.2

JeetKunDoug committed:
Merge pull request #1067 from basho/bch-merge-2.0-onto-develop-2.2

Merge 2.0 onto develop-2.2

fadushin committed:
Merge branch 'riak/2.0' into merge-2.0-into-develop-2.2

# Conflicts:
#	tests/yz_core_properties_create_unload.erl

fadushin committed:
Updated dependencies to be consistent with riak/riak_ee develop-2.2

fadushin committed:
Merge pull request #1101 from basho/merge-2.0-into-develop-2.2

Merge 2.0 into develop 2.2

fadushin committed:
dependency cleanup

Commit:
Merge pull request #1103 from basho/develop-2.2-dependency-cleanup

dependency cleanup

nickelization committed:
Add first version of verify_feature_enable_flags

Currently writes test data for all features, and tests the flag behavior
for list_buckets via protobufs. More coverage of additional features and
protocols coming soon.

nickelization committed:
Add skeleton code for testing HTTP interfaces

nickelization committed:
Add testing for HTTP list buckets

nickelization committed:
Add tests for list_keys over PB and http...

...and add test for 2i disabled over PB.

jvoegele committed:
Update verify_aae test to verify throttle config

nickelization committed:
Add tests for 2i feature flags

nickelization committed:
Add MapReduce enable/disable flag testing via HTTP

nickelization committed:
Fix typos and add test for mapred disabled via PB

nickelization committed:
Add test for mapred enabled via protocol buffers

nickelization committed:
Refactor to kill some redundancy, and log tests

tburghart committed:
added/renamed to verify_job_enable _ac (advanced.config) and _rc (riak.conf)

tburghart committed:
Refactored so reusable functions are available for other tests.
verify_listkeys and verify_job_enable_ac should be refactored to use the shared functions.
YZ flags not currently tested.

tburghart committed:
Change error message to match those for other pb job class disabled responses.

tburghart committed:
use shared macros

tburghart committed:
Get the yz switch tests working and turn them on.

fadushin committed:
Added support for specifying an upgrade callback function, which can be passed to the rt:upgrade function and which gets called after the from node is stopped and upgraded, and before the to node is started.  This allows tests to perform any additional steps as part of the upgrade.  The paths of the old and new data directories are passed elements of a prop list, which is passed into the callback as its sole argument.

jvoegele committed:
Merge pull request #1111 from basho/feature-jv-aae_throttle

Update verify_aae test to verify throttle config

tburghart committed:
tests checkpoint, implementations in job_enable_common working

jvoegele committed:
Upload debug logs to Giddyup

This can be disabled by setting the `giddyup_post_debug_logs` config
value to `false` (it is `true` by default).

tburghart committed:
Update to 2.2 switch implementation.

jvoegele committed:
Use documented function to delete existing debug log file

nickelization committed:
Update client libraries to use develop-2.2

This pulls in the latest client code, as well as newer riak_pb stuff
which is needed for the latest develop-2.2 Riak code.

nickelization committed:
Merge pull request #1130 from basho/nem-update-client-branches

Update client libraries to use develop-2.2

nickelization committed:
Update dependency versions in prep for merge

All the HLL stuff has been merged into the client develop-2.2 branches,
so we can go ahead and update these dependencies to match that.

nickelization committed:
Merge pull request #1104 from basho/feature-zl-hll_datatypes

hll tests (against develop-2.2)

nickelization committed:
Merge pull request #1122 from basho/feature-jv-giddyup_post_debug_logs

Upload debug logs to Giddyup

JeetKunDoug commented on tests/yz_crdt.erl:
```diff
@@ -113,6 +127,362 @@ confirm() ->
 
     pass.
 
+test_repeat_sets(Pid, Cluster, Bucket, Index, Key) ->
```
Some prose (comments/additional logging) in these tests would be really helpful. 430 lines of added code with almost no explanation of what's happening is tough to review and tough to maintain. A 1-2 sentence comment at the beginning of each test, plus some additional logging of what you're expecting (not just what the outcome of the search was) would be greatly appreciated and make the tests more likely to be updated/fixed in the future if someone runs into issues.

zeeshanlakhani replied:
sure. this has been waiting around in purgatory since the customer days... so that would be why it lacks it a bit. will be done shortly.

zeeshanlakhani replied:
@JeetKunDoug done.

zeeshanlakhani committed:
- add comments/logs around additional tests
- few formatting updates

fadushin committed:
Fixed yz_extractors test.

Commit:
Add back call to stop nodes before deleting index directories.
Somehow managed to pass on 2.0.7, but shouldn't have (and doesn't in 2.2.0).

jvoegele committed:
Merge pull request #1134 from basho/dr-fix-remove_index_dirs

Add back call to stop nodes before deleting index directories.

Commit:
Wait for CRDT search results, rather than failing after the first attempt.

fadushin committed:
Merge pull request #1135 from basho/dr-wait-for-crdt-results

Wait for CRDT search results, rather than failing after the first attempt.

fadushin committed:
Changed extractor test to use bad_entry stat

JeetKunDoug committed:
Merge pull request #1137 from basho/fd-extract-failure-stat

Changed extractor test to use bad_entry stat

Commit:
Disable AAE in final part of yz_extractors test to make it deterministic again.

fadushin committed:
Merge pull request #1120 from basho/fd-upgrade-downgrade-callbacks

Added support for specifying an upgrade callback function, which can …

lucafavatella commented on tests/yz_crdt.erl:
```diff
@@ -113,6 +141,387 @@ confirm() ->
+    ?assertEqual(0, length(Results2)),
+
+    lager:info("Delete key for map"),
+    ?assertEqual(ok, riakc_pb_socket:delete(Pid, Bucket, Key)),
```
Not sure it is fully in scope for this test (as this looks very yz-focused), but ideally a delete shall pass the context/vclock.

And it looks to me that at the moment it is not possible to delete a CRDT key with riakc passing the context/vclock, as the context/vclock seems not possible to be extracted (cleanly) from the value returned from riakc_pb_socket:fetch_type.

zeeshanlakhani replied:
This is deleting the entire key associated w/ the CRDT object, not removing from a set or map, which would require the context @lucafavatella.

JeetKunDoug commented:
I'm working on getting this test up and running against develop-2.2 in a separate branch. Will create a new PR once I get something running.

tburghart committed:
Add random_uniform/0,1 operations that wrap random:uniform but are always seeded with good entropy.
Putting this in a separate commit for clarity, which ideally would be a separate PR altogether except that the associated branch uses them to avoid sucking up entropy with reseeding.

tburghart committed:
Make sure the random number generator is seeded.
Print messages about why tests are skipped.
Invoke all tests, even if we know they'll be skipped for now.

tburghart committed:
Merge pull request #1126 from basho/develop-2.2-job-switches

RIAK-2583 Async Job Switches

fadushin committed:
Added logging to diagnose failure on giddyup

fadushin committed:
Added new 2.2 stats to the list of common stats.  Fixes
verify_riak_stats.

Commit:
Merge pull request #1140 from basho/fd-riak_test-verify_riak_stats

Added new 2.2 stats to the list of common stats.

Commit:
Fix 'verify_search' by enabling search on target nodes

Thanks Jason and Ted.

jvoegele committed:
Merge pull request #1142 from basho/ph-fix-turn_on_search_before_using

Fix "verify_search" by turning on search

zeeshanlakhani commented:
Closing for new pr to 2.2.

zeeshanlakhani commented:
has be replaced: https://github.com/basho/riak_test/pull/1147

Basho-JIRA commented:
Do we want to backport to 2.0.8? 

**_[posted via JIRA by Patricia Brewer]**_
